src/services/leaderboard-stat-getter.js

```diff
@@ -52,26 +60,32 @@ function LeaderboardStatGetter(model, modelRelationship, params) {
   const groupBy = `${associationAs}.${labelColumn}`;
 
   this.perform = async () => {
-    const records = await modelRelationship
-      .unscoped()
-      .findAll({
-        attributes: [
-          [model.sequelize.col(groupBy), 'key'],
-          [model.sequelize.fn(aggregate, model.sequelize.col(aggregateField)), 'value'],
-        ],
-        includeIgnoreAttributes: false,
-        include: [{
-          model,
-          attributes: [labelField],
-          as: associationAs,
-          required: true,
-        }],
-        subQuery: false,
-        group: groupBy,
-        order: [[model.sequelize.literal('value'), 'DESC']],
-        limit,
-        raw: true,
-      });
+    const { timezone } = params;
+
+    const parentSequelizeOptions = await getSequelizeOptionsForModel(parentModel, user, timezone);
+    const childSequelizeOptions = await getSequelizeOptionsForModel(childModel, user, timezone);
+
+    const records = await parentModel.findAll({
+      attributes: [
+        [childModel.sequelize.col(groupBy), 'key'],
+        [childModel.sequelize.fn(aggregate, childModel.sequelize.col(aggregateField)), 'value'],
+      ],
+      where: parentSequelizeOptions.where,
+      includeIgnoreAttributes: false,
+      include: [{
+        model: childModel,
+        attributes: [labelField],
+        as: associationAs,
+        required: true,
+        where: childSequelizeOptions.where,
```

Comment:
doesn't that crashes when the child model is using a scope on another BelongsTo than the parent model?
It looks like the issue I had on Has-many